wpilibNewCommands/src/main/native/include/frc2/command/MecanumFollowerCommand.h

```diff
@@ -0,0 +1,183 @@
+   * @param output                            The output of the velocity PIDs
+   * and feedfrowards in volts.
+   * @param requirements                      The subsystems to require.
+   */
+  MecanumFollowerCommand(
+      frc::Trajectory trajectory, std::function<frc::Pose2d()> pose,
+      frc::SimpleMotorFeedforward<units::meters> feedforward,
+      frc::MecanumDriveKinematics kinematics, frc2::PIDController xController,
+      frc2::PIDController yController,
+      frc::ProfiledPIDController thetaController,
+      units::meters_per_second_t MaxWheelVelocityMetersPerSecond,
+      std::function<frc::MecanumDriveWheelSpeeds()> currentWheelSpeeds,
+      frc2::PIDController frontLeftController,
+      frc2::PIDController rearLeftController,
+      frc2::PIDController frontRightController,
+      frc2::PIDController rearRightController,
+      std::function<void(units::voltage::volt_t, units::voltage::volt_t,
+                         units::voltage::volt_t, units::voltage::volt_t)>
+          output,
+      std::initializer_list<Subsystem*> requirements);
+
+  /**
+   * Constructs a new MecanumFollowerCommand that when executed will follow the
+   * provided trajectory. The user should implement a velocity PID on the
+   * desired output wheel velocities.
+   *
+   * <p>Note: The controllers will *not* set the output velocities to zero upon
+   * completion of the path - this is left to the user, since it is not
+   * appropriate for paths with non-stationary end-states.
+   *
+   * <p>Note 2: The rotation controller will calculate the rotation based on the
+   * final pose in the trajectory, not the poses at each time step.
+   *
+   * @param trajectory                        The trajectory to follow.
+   * @param pose                              A function that supplies the robot
+   * pose - use one of the odometry classes to provide this.
+   * @param kinematics                        The kinematics for the robot
+   * drivetrain.
+   * @param xController                      The Trajectory Tracker PID
+   * controller for the robot's x position.
+   * @param yController                      The Trajectory Tracker PID
+   * controller for the robot's y position.
+   * @param thetaController                   The Trajectory Tracker PID
+   * controller for angle for the robot.
+   * @param maxWheelVelocityMetersPerSecond   The maximum velocity of a
+   * drivetrain wheel.
+   * @param output                            The output of the velocity PID's
+   * in meters per second.
+   * @param requirements                      The subsystems to require.
+   */
+  MecanumFollowerCommand(
+      frc::Trajectory trajectory, std::function<frc::Pose2d()> pose,
+      frc::MecanumDriveKinematics kinematics, frc2::PIDController xController,
+      frc2::PIDController yController,
+      frc::ProfiledPIDController thetaController,
+      units::meters_per_second_t MaxWheelVelocityMetersPerSecond,
```

Comment:
The variable name shouldn't start with a capital letter. Furthermore, it shouldn't say the unit. That's already obvious from the type. We only did this in Java because it doesn't have a good units system.